### ISSUE TITLE
False negative on exhaustive type guard on multilevel union

### ISSUE THREAD

**Issue by mateuszkitlas** [microsoft/TypeScript#61708]
### 🔎 Search Terms

typeguard

### 🕗 Version & Regression Information

This is the behavior in every version I tried, and I reviewed the FAQ for entries about typeguard

### ⏯ Playground Link

[Playground Link](https://www.typescriptlang.org/play/?ts=5.9.0-dev.20250515#code/BQDwXABA3hDWCWA7AJpARAQzRAPhNARmgDQQYGSICuAtgQKYBOEAvrtHEqvgMYkQ9KtBsxYBKCAF4AfBFAA6BCimTJ+LLjwhFXFWsJoJAfgjbyESNp4BuAFAB6AFS2ACowD2AByYAXAJ4QAOQ8gRDI7vQAzhCI7j4Q9CDwkfHuiBD+3kEwStyY2HgG1mQUMcJMxWx4OVzofMWCZXQVrIHythAQbl6+AcGh4VExcQlJKRBpGX5ZgTUo6BqFRMXkQs2MlW3AAEwAzLsAnGK2jva2oJBzeVikq00irOxX6ES3pdTrj9Wc87z8jR8HuIpLIFLk9OoClodMpVPoiMZTPJzJZ5DYgA)

### 💻 Code

```ts
(x: { kind: "a" | "b", ab: number } | { kind: "c", c: number }) => (x.kind === "a" || x.kind === "b") ? x.ab : x.c;
```

### 🙁 Actual behavior

```
Property 'c' does not exist on type '{ kind: "a" | "b"; ab: number; } | { kind: "c"; c: number; }'.
  Property 'c' does not exist on type '{ kind: "a" | "b"; ab: number; }'.(2339)
```

### 🙂 Expected behavior

should pass

### Additional information about the issue

one-level unions work (in this case the workaround is  to move union `kind: "a" | "b"` up):
```ts
(x: { kind: "a", ab: number } | { kind: "b", ab: number } | { kind: "c", c: number }) => (x.kind === "a" || x.kind === "b") ? x.ab : x.c;
```

**Comment by MartinJohns** [comment]
Duplicate of #31404.
